src/panels/lovelace/badges/types.ts

```diff
@@ -25,3 +26,15 @@ export interface StateLabelBadgeConfig extends LovelaceBadgeConfig {
   hold_action?: ActionConfig;
   double_tap_action?: ActionConfig;
 }
+
+export interface EntityBadgeConfig extends LovelaceBadgeConfig {
+  type: "entity";
+  entity?: string;
+  name?: string;
+  icon?: string;
```

Comment:
The `StateLabelBadgeConfig` supported `image`, should we keep that around in the new one?

Maybe also migrate the option `show_name` to a `display_type`?

Reply:
I will make another PR to properly migrate the state label badge